src/apm-cli.coffee

```diff
@@ -100,34 +100,49 @@ printVersions = (args, callback) ->
 
   getPythonVersion (pythonVersion) ->
     git.getGitVersion (gitVersion) ->
-      if args.json
-        versions =
-          apm: apmVersion
-          npm: npmVersion
-          node: nodeVersion
-          python: pythonVersion
-          git: gitVersion
-          nodeArch: process.arch
-        if config.isWin32()
-          versions.visualStudio = config.getInstalledVisualStudioFlag()
-        console.log JSON.stringify(versions)
-      else
-        pythonVersion ?= ''
-        gitVersion ?= ''
-        versions =  """
-          #{'apm'.red}  #{apmVersion.red}
-          #{'npm'.green}  #{npmVersion.green}
-          #{'node'.blue} #{nodeVersion.blue} #{process.arch.blue}
-          #{'python'.yellow} #{pythonVersion.yellow}
-          #{'git'.magenta} #{gitVersion.magenta}
-        """
-
-        if config.isWin32()
-          visualStudioVersion = config.getInstalledVisualStudioFlag() ? ''
-          versions += "\n#{'visual studio'.cyan} #{visualStudioVersion.cyan}"
-
-        console.log versions
-      callback()
+      getAtomVersion (atomVersion) ->
+        if args.json
+          versions =
+            apm: apmVersion
+            npm: npmVersion
+            node: nodeVersion
+            atomVersion: atomVersion
+            python: pythonVersion
+            git: gitVersion
+            nodeArch: process.arch
+          if config.isWin32()
+            versions.visualStudio = config.getInstalledVisualStudioFlag()
+          console.log JSON.stringify(versions)
+        else
+          pythonVersion ?= ''
+          gitVersion ?= ''
+          atomVersion ?= ''
+          versions =  """
+            #{'apm'.red}  #{apmVersion.red}
+            #{'npm'.green}  #{npmVersion.green}
+            #{'node'.blue} #{nodeVersion.blue} #{process.arch.blue}
+            #{'atom'.cyan} #{atomVersion.cyan}
+            #{'python'.yellow} #{pythonVersion.yellow}
+            #{'git'.magenta} #{gitVersion.magenta}
+          """
+
+          if config.isWin32()
+            visualStudioVersion = config.getInstalledVisualStudioFlag() ? ''
+            versions += "\n#{'visual studio'.cyan} #{visualStudioVersion.cyan}"
+
+          console.log versions
+        callback()
+
+getAtomVersion = (callback) ->
+  config.getResourcePath (resourcePath) ->
+    # workaround if app.asar does not exist. see: https://github.com/atom/atom/issues/6604
+    if not fs.existsSync(resourcePath)
+      resourcePath = resourcePath.replace('.asar', path.sep)
+    try
+      {version} = require(path.join(resourcePath, 'package.json')) ? {}
```

Comment:
It would probably be better to return `unknown` rather than an empty object.

Reply:
Please see 99bec866c36a24f7f9bfb2ae6d523a9df9538200